functions/index.js

```diff
@@ -50,6 +50,24 @@ exports.updateWhitelist = functions.firestore
   .document('/addendums/{id}')
   .onCreate(clalib.updateWhitelist)
 
+/**
+ * Periodically re-generates all whitelists in case of one-off errors with
+ * snapshot-based updateWhitelist.
+ */
+exports.reconcileWhitelist = functions.pubsub
+  .schedule('every 15 minutes')
```

Comment:
Could this be done after any change to an agreement/addendum, rather than scheduled?

Reply:
We already do this (see function above `updateWhitelist`). This is to correct potential one-off errors with the update logic triggered for each addendum create event. In other words, `updateWhitelist` attempts at updating the whitelist right after an agreement is updated. `reconcileWhitelist` guarantees that in case of transient errors, the whitelist will eventually reach the correct state.